www/perl-lib/SMT/Rest/SCCAPIv1.pm

```diff
@@ -10,14 +10,30 @@ use JSON;
 
 use SMT::Utils;
 use SMT::Client;
-use SMT::Registration;
-
+use SMT;
 
 sub new
 {
     my($class, $r) = @_;
     my $self = $class->SUPER::new($r);
 
+    $self->{regsharing} = SMT::Utils::hasRegSharing($r);
+    if ( $self->{regsharing} eq 'true' ) {
+        eval
+        {
+            #require 'SMT/RegistrationSharing.pm';
+	    require SMT::RegistrationSharing;
+        };
+        if ($@)
+        {
+            my $msg = 'Failed to load registration sharing module '
+                . '"SMT/RegistrationSharing.pm"'
+                . "\n$@";
+            $r->log_error($msg);
+            $self->{regsharing} = 'false';
+        }
+    }
```

Comment:
I think this block is something for Base.pm. 
Base is the right class for initializations like this.